src/app/core/services/vault/vault.service.ts

```diff
@@ -189,27 +212,42 @@ export class VaultService {
       try {
         const decryptedState = await encryptor.decrypt(vaultPassword, encryptedState);
         if (this.isValidState(decryptedState)) {
-          this.localStorageStore.putState(decryptedState);
+          let newState = decryptedState;
+          let fctAddressesRequestWhitelistedDomains = [];
+          let ecAddressesRequestWhitelistedDomains = [];
+
+          if (decryptedState.version === environment.localStorageVersion) {
+            fctAddressesRequestWhitelistedDomains = JSON.parse(decryptedState.fctAddressesRequestWhitelistedDomains);
+            ecAddressesRequestWhitelistedDomains = JSON.parse(decryptedState.ecAddressesRequestWhitelistedDomains);
+          } else {
+            newState = Object.assign({}, decryptedState, {
+              version: environment.localStorageVersion,
+              fctAddressesRequestWhitelistedDomains: JSON.stringify([]),
+              ecAddressesRequestWhitelistedDomains: JSON.stringify([])
+            });
+          }
```

Comment:
Also when trying to restore your vault using a backup with the older version, the state will be updated again.